src/debot/term_encryption_box.rs

```diff
@@ -0,0 +1,240 @@
+        Ok(EncryptionBoxInfo {
+            hdpath: Some(String::from("0")),
+            algorithm: Some("ChaCha20".to_string()),
+            options: Some(json!({"nonce": self.nonce.clone()})),
+            public: None,
+        })
+    }
+    async fn encrypt(&self, data: &String) -> ClientResult<String> {
+        Ok(chacha20(
+            self.client.clone(),
+            ParamsOfChaCha20 {
+                data: base64::encode(data),
+                key: self.key.clone(),
+                nonce: self.nonce.clone(),
+            },
+        )
+        .unwrap()
+        .data)
+    }
+    async fn decrypt(&self, data: &String) -> ClientResult<String> {
+        Ok(chacha20(
+            self.client.clone(),
+            ParamsOfChaCha20 {
+                data: data.clone(),
+                key: self.key.clone(),
+                nonce: self.nonce.clone(),
+            },
+        )
+        .unwrap()
+        .data)
+    }
+}
+
+#[async_trait::async_trait]
+impl ton_client::crypto::EncryptionBox for NaClBox {
+    async fn get_info(&self) -> ClientResult<EncryptionBoxInfo> {
+        Ok(EncryptionBoxInfo {
+            hdpath: Some(String::from("0")),
+            algorithm: Some("NaclBox".to_string()),
+            options: Some(json!({
+                "their_public": self.their_pubkey.clone(),
+                "nonce": self.nonce.clone()})),
+            public: None,
+        })
+    }
+    async fn encrypt(&self, data: &String) -> ClientResult<String> {
+        Ok(nacl_box(
+            self.client.clone(),
+            ParamsOfNaclBox {
+                decrypted: base64::encode(data),
+                nonce: self.nonce.clone(),
+                their_public: self.their_pubkey.clone(),
+                secret: self.secret.clone(),
+            },
+        )
+        .unwrap()
+        .encrypted)
+    }
+    async fn decrypt(&self, data: &String) -> ClientResult<String> {
+        Ok(nacl_box_open(
+            self.client.clone(),
+            ParamsOfNaclBoxOpen {
+                encrypted: data.clone(),
+                nonce: self.nonce.clone(),
+                their_public: self.their_pubkey.clone(),
+                secret: self.secret.clone(),
+            },
+        )
+        .unwrap()
+        .decrypted)
+    }
+}
+
+pub(super) struct TerminalEncryptionBox {
+    handle: EncryptionBoxHandle,
+    box_type: EncryptionBoxType,
+    client: TonClient,
+}
+
+impl TerminalEncryptionBox {
+    pub async fn new(params: ParamsOfTerminalEncryptionBox) -> Self {
+        let key: String;
+
+        {
+            let stdio = io::stdin();
+            let mut reader = stdio.lock();
+            let mut writer = io::stdout();
+            let enter_str = "enter seed phrase or path to keypair file";
+            let value = input(enter_str, &mut reader, &mut writer);
+            let pair = load_keypair(&value).map_err(|e| e.to_string());
+            key = format!("{:064}", pair.unwrap().secret);
```

Comment:
It's bad to use unwrap here. It is common to have errors here so we should propagate it back to the caller.